### DISCUSSION TITLE
Adafruit Metro ESP32-S2 which method?

### DISCUSSION THREAD

**Discussion by hallard** [Makuna/NeoPixelBus#497] [Q&A] [answered]
Sorry to ask guys but today I'm playing with Adafruit Metro ESP32-S2 board.

Neopixel is on GPIO45 

![image](https://user-images.githubusercontent.com/2471931/124775376-dcc81a80-df3e-11eb-8911-4dbe0d68dfb1.png)

I'm using latest arduino [ESP32 2.0.0-alpha1](https://github.com/espressif/arduino-esp32/tree/2.0.0-alpha1) but I'm unable to compile basic example (I already used this lib in the past with several targets without any issue) so my guess it's related to ESP32-S2

I tried all the declaration (not using strip yet, just instantiate it), so if someone knows what method I should use on this board it will be awesome.

`NeoPixelBus<NeoGrbFeature, NeoEsp32BitBang800KbpsMethod> strip(1, 45);`
```
/Users/charles/OneDrive/devt/Arduino/libraries/NeoPixelBus_by_Makuna/src/internal/NeoEsp32RmtMethod.h:873:9: error: 'NeoEsp32Rmt3Tm1914Method' does not name a type; did you mean 'NeoEsp32Rmt3Tm1814Method'?
 typedef NeoEsp32Rmt3Tm1914Method NeoTm1914Method;
         ^~~~~~~~~~~~~~~~~~~~~~~~
         NeoEsp32Rmt3Tm1814Method
```

`NeoPixelBus<NeoGrbFeature, NeoEsp32Rmt0Ws2812xMethod> strip(1, 45);`
```

/Users/charles/OneDrive/devt/Arduino/libraries/NeoPixelBus_by_Makuna/src/internal/NeoEsp32RmtMethod.h:873:9: error: 'NeoEsp32Rmt3Tm1914Method' does not name a type; did you mean 'NeoEsp32Rmt3Tm1814Method'?
 typedef NeoEsp32Rmt3Tm1914Method NeoTm1914Method;
         ^~~~~~~~~~~~~~~~~~~~~~~~
         NeoEsp32Rmt3Tm1814Method
```

`NeoPixelBus<NeoGrbFeature, NeoEsp32Rmt3Ws2812xMethod> strip(1, 45);`
```
/Users/charles/OneDrive/devt/Arduino/libraries/NeoPixelBus_by_Makuna/src/internal/NeoEsp32RmtMethod.h:873:9: error: 'NeoEsp32Rmt3Tm1914Method' does not name a type; did you mean 'NeoEsp32Rmt3Tm1814Method'?
 typedef NeoEsp32Rmt3Tm1914Method NeoTm1914Method;
         ^~~~~~~~~~~~~~~~~~~~~~~~
         NeoEsp32Rmt3Tm1814Method
```

`NeoPixelBus<NeoGrbFeature, NeoEsp32I2s0800KbpsMethod> strip(1, 45);`
```
/Users/charles/OneDrive/devt/Arduino/libraries/NeoPixelBus_by_Makuna/src/internal/NeoEsp32RmtMethod.h:873:9: error: 'NeoEsp32Rmt3Tm1914Method' does not name a type; did you mean 'NeoEsp32Rmt3Tm1814Method'?
 typedef NeoEsp32Rmt3Tm1914Method NeoTm1914Method;
         ^~~~~~~~~~~~~~~~~~~~~~~~
         NeoEsp32Rmt3Tm1814Method
```

**Comment by Makuna** [comment, marked as answer]
known bug that has been fixed but not in an official release yet.

https://github.com/Makuna/NeoPixelBus/issues/494

Pick up the current GitHub version of NeoPixelBus.  The readme has the simple instructions to accomplish this.

Confirm it works for you and I will push out a release. 

BTW, while a single pixel may work reliably using the BitBang method, I never recommend it.  See [ESP32 Methods Wiki Page](https://github.com/Makuna/NeoPixelBus/wiki/ESP32-NeoMethods) for details on specific methods for the ESP32 platform.
AND, since this onboard NeoPixel is on pin 46, then BitBang can't be used as it only supports pins below 32.  Just use the default method names that aren't specific to hardware/platform and the best one will be used.  Like `NeoWs2812xMethod` and you will be fine.

**Reply by hallard** [reply]
> BTW, while a single pixel may work reliably using the BitBang method, I never recommend it. See ESP32 Methods Wiki Page for details on specific methods for the ESP32 platform.

Sure was aware of, was just testing.

> Pick up the current GitHub version of NeoPixelBus. The readme has the simple instructions to accomplish this.

got `NeoESP32RmtMethod.h` from git, works fine.

Confirmed to work with 
`NeoPixelBus<NeoGrbFeature, NeoEsp32I2s0800KbpsMethod> strip(1, 45);`

Just to mention again for users, take care ESP32-S2 does not have 2nd I2S so 
`NeoPixelBus<NeoGrbFeature, NeoEsp32I2s1800KbpsMethod> strip(1, 45);`
won't compile :-)

Thanks @Makuna as usual for you support.
